cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -1123,9 +1124,10 @@ impl ABIMachineSpec for AArch64MachineDeps {
         }
     }
 
-    fn get_regs_clobbered_by_call(call_conv: isa::CallConv) -> PRegSet {
+    fn get_regs_clobbered_by_call(call_conv: isa::CallConv, is_exception: bool) -> PRegSet {
         match call_conv {
             isa::CallConv::Winch => WINCH_CLOBBERS,
+            _ if is_exception => ALL_CLOBBERS,
```

Comment:
Why are all registers clobbered for `try_call`? For regular systemv unwinding, all callee saved registers are preserved on the unwind path too.

Reply:
Ah, yes, this came out of our discussions regarding the Wasmtime side of the implementation: we didn't want to have to require all registers to be saved and accessible in all trampolines out to the host as a starting-point for the unwind register state (otherwise we are penalizing. all existing programs when the feature is turned on); and we didn't necessarily want to have the unwind-info be load-bearing for correctness. With this approach we can restart at the catch-point without having to set up any register state other than the payload.

This could definitely become a codegen flag for use in native code; not doing that is just an oversight in the initial PR, sorry. I'd be happy to review a PR for that!